tests/components/mobile_app/test_http_api.py

```diff
@@ -63,6 +64,86 @@ async def test_registration(
     )
 
 
+async def test_reregistration(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    hass_admin_user: MockUser,
+) -> None:
+    """Test that reregistrations overwrite the current entry."""
+    await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+    extra_config_entry = MockConfigEntry(
```

Comment:
This is the first config entry, it's not extra.

Reply:
Yep, I am still tinkering around with the code, how the test is setup it should create extra entries, which we should prevent with the new config_entries logic erik is implementing